diesel/src/query_dsl/save_changes_dsl.rs

```diff
@@ -1,72 +1,66 @@
 use associations::HasTable;
-use backend::SupportsReturningClause;
-use connection::Connection;
 use helper_types::*;
 use query_builder::{AsChangeset, IntoUpdateTarget};
 use query_dsl::*;
-use query_source::Queryable;
 use result::QueryResult;
-use types::HasSqlType;
 #[cfg(any(feature = "sqlite", feature = "mysql"))]
 use associations::Identifiable;
 
-pub trait SaveChangesDsl<Conn, ST> where
-    Conn: Connection,
-    Conn::Backend: HasSqlType<ST>,
-{
-    fn save_changes<T>(self, connection: &Conn) -> QueryResult<T> where
-        T: Queryable<ST, Conn::Backend>;
+pub trait InternalSaveChangesDsl<Conn, T>: Sized {
+    fn internal_save_changes(self, connection: &Conn) -> QueryResult<T>;
 }
 
-impl<T, ST, Conn> SaveChangesDsl<Conn, ST> for T where
-    Conn: Connection,
-    Conn::Backend: HasSqlType<ST> + SupportsReturningClause,
+impl<T, U, Conn> InternalSaveChangesDsl<Conn, U> for T where
     T: Copy + AsChangeset<Target=<T as HasTable>::Table> + IntoUpdateTarget,
-    Update<T, T>: LoadDsl<Conn, SqlType=ST>,
+    Update<T, T>: LoadDsl<Conn> + LoadQuery<Conn, U>,
```

Comment:
`LoadQuery` implies `LoadDsl`, but we need to explicitly mention it here in order to make this impl disjoint from the sqlite/mysql ones.